src/Generators/Classes/ParsePermissions.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace DeInternetJongens\LighthouseUtils\Generators\Classes;
+
+use DeInternetJongens\LighthouseUtils\Models\GraphQLSchema;
+use GraphQL\Language\AST\ArgumentNode;
+use GraphQL\Language\AST\DirectiveNode;
+use GraphQL\Language\AST\DocumentNode;
+use GraphQL\Language\AST\NodeList;
+use GraphQL\Language\AST\ObjectTypeDefinitionNode;
+use GraphQL\Language\Parser;
+use GraphQL\Language\Source;
+
+class ParsePermissions
+{
+    /**
+     * @param string $fileContents
+     * @return array
+     * @throws \GraphQL\Error\SyntaxError
+     */
+    public function register(string $fileContents): array
+    {
+        $parser = $this->parseGraphQLSchema($fileContents);
+
+        if (empty($parser->definitions[0])) {
+            return [];
+        }
+
+        /** @var ObjectTypeDefinitionNode $firstNode */
+        $wrapper = $parser->definitions[0];
+
+        $rows = [];
+
+        /** @var NodeList $field */
+        foreach ($wrapper->fields as $field) {
+            $arguments = [];
+
+            $model = $field->type->type->name->value ?? $field->type->name->value;
```

Comment:
Can we be sure that any of these two values will always be set?

Reply:
Yes, because it throws a GraphQL syntax error if its not